src/test/java/seedu/address/logic/parser/AddTaskCommandParserTest.java

```diff
@@ -0,0 +1,107 @@
+import static seedu.address.logic.commands.CommandTestUtil.TASK_DESCRIPTION_TASK2;
+import static seedu.address.logic.commands.CommandTestUtil.TASK_NAME_TASK1;
+import static seedu.address.logic.commands.CommandTestUtil.TASK_NAME_TASK2;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TASK_DESCRIPTION;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TASK_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TASK_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TASK_NAME;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static seedu.address.testutil.TypicalTasks.TASK1;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.AddTaskCommand;
+import seedu.address.model.task.Task;
+import seedu.address.model.task.TaskName;
+import seedu.address.testutil.TaskBuilder;
+
+public class AddTaskCommandParserTest {
+    private AddTaskCommandParser parser = new AddTaskCommandParser();
+
+    @Test
+    public void parse_allFieldsPresent_success() {
+        Task expectedTask = new TaskBuilder(TASK1).build();
+
+        // whitespace only preamble
+        assertParseSuccess(parser, PREAMBLE_WHITESPACE
+                + TASK_NAME_TASK1 + TASK_DESCRIPTION_TASK1, new AddTaskCommand(expectedTask));
+    }
+
+    @Test
+    public void parse_repeatedValue_failure() {
+        String validExpectedTaskString = TASK_NAME_TASK1 + TASK_DESCRIPTION_TASK1;
+
+        // multiple names
+        assertParseFailure(parser, TASK_NAME_TASK2 + validExpectedTaskString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_TASK_NAME));
+
+        // multiple desscription
+        assertParseFailure(parser, TASK_DESCRIPTION_TASK2 + validExpectedTaskString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_TASK_DESCRIPTION));
+
+        // invalid value followed by valid value
+
+        // invalid name
+        assertParseFailure(parser, INVALID_TASK_NAME + validExpectedTaskString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_TASK_NAME));
+
+        // invalid description
+        assertParseFailure(parser, INVALID_TASK_DESCRIPTION + validExpectedTaskString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_TASK_DESCRIPTION));
+
+
+        // valid value followed by invalid value
+
+        // invalid name
+        assertParseFailure(parser, validExpectedTaskString + INVALID_TASK_NAME,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_TASK_NAME));
+
+        // invalid description
+        assertParseFailure(parser, validExpectedTaskString + INVALID_TASK_DESCRIPTION,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_TASK_DESCRIPTION));
+
+    }
+
```

Comment:
Im assuming parse_optionalFieldsMissing_success or failure isnt needed here.
I was thinking maybe the task description can be optional? as the task name sometimes explains itself.